plugins-ui/src/modules/shared/wallet/Wallet.tsx

```diff
@@ -78,24 +77,31 @@ const Wallet = () => {
       // 3. Store public key in localStorage
       localStorage.setItem("publicKey", publicKey);
 
-      // 4. Generate hex message for signing
-      const signingMessage = "Sign into Vultisig App Store\n\nAddress: " + walletAddress;
+      // 4. Generate nonce and expiry timestamp
+      const nonce = ethers.hexlify(ethers.randomBytes(16));
+      const expiryTime = new Date(Date.now() + 15 * 60 * 1000).toISOString(); // 15 minutes from now
+
+      // 5. Generate hex message for signing
+      const signingMessage = `Sign into Vultisig App Store
++Nonce: ${nonce}
```

Comment:
can we please use structure data?  Construct a json object

Reply:
done